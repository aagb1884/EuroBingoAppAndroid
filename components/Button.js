import { StyleSheet, View, Pressable, Text } from 'react-native';


export default function Button({ isActive, cliche, onPress, index}) {

  return (
    <View style={[styles.buttonContainer]}>
    <Pressable
      style={[isActive ? styles.active : styles.button]}
      onPress={onPress}
      testID={`card-button-${index}`}
    >
     
      <Text style={[isActive ? styles.activeText : styles.buttonLabel]}>{cliche}</Text>
    </Pressable>
  </View>
  );
}


const styles = StyleSheet.create({
  buttonContainer: {
    width: 340,
    height: "auto",
    marginHorizontal: 20,
    marginTop: 10,
    marginBottom: 10,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 3,
    borderRadius: 18,
    },
    
  button: {
    backgroundColor: "#FF1361",
    borderRadius: 10,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    borderWidth: 1,
    borderColor: '#000',
    padding: 10,
    
  },
   buttonLabel: {
    color: '#25292e',
    fontSize: 20,
    fontWeight: "bold",
    textAlign: "center"
  },
   active: {
   borderRadius: 10,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    borderWidth: 1,
    borderColor: '#000',
    padding: 10,
    margin: 5,
    backgroundColor: '#231557', 
  },
   activeText: {
       color: '#fff', 
       fontSize: 20,
       fontWeight: "bold",
       textAlign: "center"
  },
});
