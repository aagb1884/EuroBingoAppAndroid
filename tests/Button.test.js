import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

describe('<Button />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
   
  it('should call onPress when the button is pressed', () => {
   
    const onPressMock = jest.fn();

    const component = renderer.create(
      <Button
      isActive={true}
      cliche={"example cliche"}
      onPress={onPressMock}
      key={1}
      />
    )

      const buttonInstance = component.root.findByProps({cliche: 'example cliche'});

      buttonInstance.props.onPress();
   
      expect(onPressMock).toHaveBeenCalled();
});

});