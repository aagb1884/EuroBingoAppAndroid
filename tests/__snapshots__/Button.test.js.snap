// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 18,
        "height": "auto",
        "justifyContent": "center",
        "marginBottom": 10,
        "marginHorizontal": 20,
        "marginTop": 10,
        "padding": 3,
        "width": 340,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FF1361",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        },
      ]
    }
    testID="card-button-undefined"
  >
    <Text
      style={
        [
          {
            "color": "#25292e",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    />
  </View>
</View>
`;
